fdbrpc/CMakeLists.txt

```diff
@@ -104,6 +104,10 @@ if(WIN32)
   add_dependencies(fdbrpc_sampling_actors fdbrpc_actors)
 endif()
 
+add_flow_target(LINK_TEST NAME fdbrpclinktest SRCS ${FDBRPC_SRCS} LinkTest.cpp)
```

Comment:
This is a good idea. I tried moving the chunk that generates the executable in the `add_flow_target` function into the conditional that creates a static library, so it would always create both a static library and a link test executable. Unfortunately, my cmake skills are not great and it didn't seem to want to actually make an executable for some reason.

Reply:
I was forgetting to add the `LinkTest.cpp` file to the sources 🤦 . However, there is still an issue here because the new target doesn't have the same library dependencies as the original target, and attempting to link using the original target doesn't pick up any of the private interfaces. For example, this is the gist of what I added:

```
add_executable(${AFT_NAME}linktest ${sources} ${AFT_ADDL_SRCS} ${CMAKE_CURRENT_SOURCE_DIR}/${src}/LinkTest.cpp)
target_link_libraries(${AFT_NAME}linktest PRIVATE ${AFT_NAME})
```

but the `flow` library has the following:

```
target_link_libraries(${ft} PRIVATE stacktrace)
```

and so when building the `flowlinktest` executable none of the `stacktrace` symbols are found. This is fixable by either linking `stacktrace` as public (basically all libraries would need to be linked as public), or by manually adding the `stacktrace` library as a target of `flowlinktest` in `flow/CMakeLists.txt`. But then we are kind of back at square one where it requires additional work for each new target (although there is the added benefit of forcing these libraries to be added).

Overall, I'm not sure this added complexity is worth it at this point.